### PR TITLE
feature(widgets): customizable texts

### PR DESCRIPTION
**Pull request by digorath**
## Changes proposed in this PR:

- added option to customize texts serversided
- exchanged widget touchable with touchable opacity
  - with no padding horizontally the touch shadow looked wrong

SVA-154

### CODE REVIEW

File: src/components/widgets/EventWidget.tsx [hunk 1]
```diff
@@ -35,7 +31,7 @@ export const EventWidget = ({ navigation }: Props) => {
 
   const onPress = useCallback(() => {
     navigation.navigate('Index', {
-      title: 'Veranstaltungen',
+      title: text ?? texts.homeTitles.events,
```
Comment (donni106) [inline src/components/widgets/EventWidget.tsx:34]: 👍

File: src/components/widgets/WeatherWidget.tsx [hunk 2]
```diff
@@ -30,8 +30,8 @@ export const WeatherWidget = ({ navigation }: { navigation: NavigationScreenProp
   useHomeRefresh(refetch);
 
   return (
-    <Touchable
-      onPress={() => navigation?.navigate('Weather', { weatherData: { icon, temperature } })}
+    <TouchableOpacity
+      onPress={() => navigation?.navigate('Weather', { title: text ?? texts.screenTitles.weather })}
```
Comment (donni106) [inline src/components/widgets/WeatherWidget.tsx:34]: the other widgets have `useCallback` navigation onpresses, right? but here not?!

File: src/screens/HomeScreen.js [hunk 3]
```diff
@@ -148,7 +148,7 @@ export const HomeScreen = ({ navigation }) => {
           publicJsonFile="homeCarousel"
           refreshTimeKey="publicJsonFile-homeCarousel"
         />
-        <Widgets navigation={navigation} widgets={widgets} />
+        <Widgets navigation={navigation} widgetConfigs={widgets} />
```
Comment (donni106) [inline src/screens/HomeScreen.js]: lets name the `widgets` also `widgetConfigs` here in `HomeScreen` for clearer reference?

Comment (digorath) [reply src/screens/HomeScreen.js]: done

File: src/components/widgets/WeatherWidget.tsx (hunk 2 above)
Comment (digorath) [reply src/components/widgets/WeatherWidget.tsx]: i missed that back when i implemented it i guess. it is inside a useCallback before now.

i also fixed the dependencies of all the onpress callbacks of the widgets

**Review by donni106 — APPROVED** [review]

**Pull request merged**